src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
 
-interface ListItemProps {
-  cv: string;
-}
-const ListItem = ({ cv }: ListItemProps): JSX.Element => {
-  // todo: implement the logic
-  const handleDownload = (): void => {
-    const x = 'Downloaded';
-    // eslint-disable-next-line no-console
-    console.log(x);
+import { APP_DATA_TYPES } from '../../../config/appDataTypes';
+import { useAppDataContext } from '../../context/AppDataContext';
+import { TEMPLATES } from '../read/cvForm/constants';
+import FirstTemplate from '../read/cvForm/templates/FirstTemplate';
+import DataFilter from './DataFilter';
+import { CandidateRateObj } from './types';
+
+const BuilderView: FC = () => {
+  const { postAppData, patchAppData, appDataArray } = useAppDataContext();
+  const handlePatch = (
+    dataObj: AppData['id'],
+    newData: CandidateRateObj,
+  ): void => {
+    patchAppData({ id: dataObj, data: newData });
   };
-  // todo: implement the logic
-  const handleView = (): void => {
-    const x = 'Viewed';
-    // eslint-disable-next-line no-console
-    console.log(x);
+  const handlePost = (newdata: CandidateRateObj): void => {
+    postAppData({ data: newdata, type: APP_DATA_TYPES.CANDIDATE_RATE_INFO });
+  };
+  const groupedData = appDataArray.groupBy((data) => data.type).toJS();
+  const [selectedType, setSelectedType] = useState('');
+  const handleTypeChange = (value: string): void => {
+    setSelectedType(value);
+  };
+  const dataArray: any[] = Object.keys(groupedData)
+    .filter((key) => key === selectedType)
+    .map((key) => groupedData[key]);
+  const dimensions = [
+    { value: 'mock_type', label: 'Mock Info' },
+    { value: 'personalInfo', label: 'Personal Info' },
+    { value: 'educationInfo', label: 'Education Info' },
+    { value: 'workInfo', label: 'Work Info' },
+    { value: 'skillsInfo', label: 'Skills Info' },
+    { value: 'projectsInfo', label: 'Projects Info' },
+    { value: 'motivationInfo', label: 'Self-Motivation' },
+    { value: 'referencesInfo', label: 'References Info' },
+    { value: 'cvStatusInfo', label: '' },
+  ];
+
+  // Construct the 'cv' object with filtered and grouped data
+  const candidatesCv: { [key: string]: { [key: string]: any } } = {};
+
+  dimensions.forEach((dimension) => {
+    const type = dimension.value;
+    const dataObj = Object.keys(groupedData)
+      .filter((key) => key === type)
+      .map((key) => groupedData[key]);
+
+    const typeDataByMemberId = DataFilter({
+      dataObject: dataObj,
+      targetType: type,
+    });
+
+    Object.keys(typeDataByMemberId).forEach((memberId) => {
+      if (memberId in candidatesCv) {
+        candidatesCv[memberId][type] = typeDataByMemberId[memberId].map(
+          (item) => item.data,
+        );
+      } else {
+        candidatesCv[memberId] = {
+          [type]: typeDataByMemberId[memberId].map((item) => item.data),
+        };
+      }
+    });
+  });
+
+  const [ratings, setRatings] = useState<{ [key: string]: number | null }>({});
```

Comment:
Where do you get the ratings from the backend (appData) ? It looks like you are saving them but never actually using the ones you have already saved.

Reply:
I'm handling the rating inside the handleRatingChange function, where I would check for existing appData for it or not, and use `patchAppData` & `postAppData`, and I hope this is the solution for what you have mentioned

Reply:
Yes I got the part here you handle adding a new rating, posting it to the api and adding it to the state. But what about ratings that are already in the backend when you open the app ? you never get them. Your interface would show no ratings if a user had rated candidates previously and opens the interface a day later for example.

Reply:
Ah, yeh I see what you mean, I will work on fixing this.